### PR TITLE
[PYIC-2839] Update retrieve-cri-credential for initial F2F response

### PR DESCRIPTION
**Pull request by therealslimsladey**
### What changed
retrieve-cri-credential updated to handle the initial 'pending` response from F2F CRI

### Why did it change
Integration with F2F CRI

### Issue tracking
- [PYIC-2839](https://govukverify.atlassian.net/browse/PYIC-2839)

### Environment variables or secrets
- No environment variables or secrets were added or changed

### Other considerations

[PYIC-2839]: https://govukverify.atlassian.net/browse/PYIC-2839?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by shivanshuit914 — DISMISSED** [review]

File: lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java
Comment (chris-dco) [inline lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java:220]: Could do with a test to ensure that this is correctly handled. Currently Sonarqube indicates that this is untested.

File: libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/VerifiableCredentialService.java
Comment (chris-dco) [inline libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/VerifiableCredentialService.java]: I think this is the only place in the code where `var` is used. Should this be `VerifiableCredentialResponse`?

File: libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/VerifiableCredentialService.java
Comment (chris-dco) [inline libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/VerifiableCredentialService.java:153]: Is it worth adding a test which executes this if block? Currently Sonarqube indicates it's not tested.

File: lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java
Comment (chris-dco) [inline lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java:212]: Does this require an audit event?

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/CriResponseService.java
Comment (chris-dco) [inline lib/src/main/java/uk/gov/di/ipv/core/library/service/CriResponseService.java:53]: Should this store the dateCreated and/or expirationTime?

File: libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java
Comment (chris-dco) [inline libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java:14]: Will there be a state when this is approved or one where it's rejected?

File: libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java
Comment (blakeyp) [inline libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java:22]: I'm guessing the plan is to update this to CREATED when the actual VC is received?

**Review by blakeyp — DISMISSED** [review]

File: lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java
Comment (therealslimsladey) [reply lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java:220]: Good shout. Test added

File: libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/VerifiableCredentialService.java
Comment (therealslimsladey) [reply libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/VerifiableCredentialService.java]: I've typed all the locals rather than using var :)

File: libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/VerifiableCredentialService.java
Comment (therealslimsladey) [reply libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/VerifiableCredentialService.java:153]: Good shout. I've added a test.

File: lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java
Comment (therealslimsladey) [reply lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java:212]: It may well do, but I've added that as a separate subtask (https://govukverify.atlassian.net/browse/PYIC-2911) which may well get pushed to the fast-follow epic

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/CriResponseService.java
Comment (therealslimsladey) [reply lib/src/main/java/uk/gov/di/ipv/core/library/service/CriResponseService.java:53]: Yup, my bad: I missed the dateCreated (now added). We have no immediate plans for the expirationTime so I've removed it from the class - it was there from the ctrl-c/v of the user credentials store. The 14-day table ttl will handle the aging out of the pending credentials.

File: libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java
Comment (therealslimsladey) [reply libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java:14]: Not to my knowledge - the ADR only indicates pending and created responses. I suspect you may well be correct going forward.

File: libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java
Comment (therealslimsladey) [reply libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java:22]: There's no immediate plan to fixup the CRI response table when the actual VC is received. The CRI response item will hang around for its 14 day TTL and then gracefully disappear.

Comment (blakeyp) [reply libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/domain/VerifiableCredentialStatus.java:22]: It would help us to easily work out whether an incoming user still has a pending F2F request (https://govukverify.atlassian.net/browse/PYIC-2846). But for now we can just check that there's no F2F VC for them in the actual VC store

**Review by chris-dco — DISMISSED** [review]
Happy with the changes. Good to wait for @blakeyp to approve before merging.

**Review by chris-dco — APPROVED** [review]

**Pull request merged**
